pallets/author-inherent/src/lib.rs

```diff
@@ -117,7 +117,7 @@ pub mod pallet {
 		/// but before transactions are executed.
 		// This should go into on_post_inherents when it is ready https://github.com/paritytech/substrate/pull/10128
 		// TODO better weight. For now we just set a somewhat conservative fudge factor
-		#[pallet::weight((10 * T::DbWeight::get().write, DispatchClass::Mandatory))]
+		#[pallet::weight((3 * T::DbWeight::get().read + 11 * T::DbWeight::get().write, DispatchClass::Mandatory))]
```

Comment:
I understand that this is a fudge factor, but it seems extremely high. What's the rationale for them?

Reply:
It was 10 writes before and now it is 11 writes. The `HighestSlotSeen` added 1 more write.

I do not know why the fudge factor was 7 writes, it isn't benchmarked so the author (probably @JoshOrndorff ) was presumably being conservative.

Reply:
Yeah, it was me that added that fudge factor, and it was exactly as Amar says. I just wanted something quick and safe that was better than `0`.

Reply:
This is tricky because it calls the `SlotBeacon` which can have any implementation it wants. I think this problem has been solved in substrate land. It's the problem of weighing functionality that is abstracted behind a trait. How do they do it for all the pallets that depend on `Currency`?

Reply:
@girazoki told me it's best to return (ReturnValue, WeightConsumed) in that case for the trait that is getting the value. So we'd need to change the SlotBeacon trait to return this tuple and change the impl to return the weight consumed too.

Reply:
I am in the process of benchmarking this, I think I have almost everything set except for https://github.com/paritytech/cumulus/pull/1187.